File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    result = []
    l = len(elements)
    elements = elements + elements
    # 최대 길이
    for limit in range(1, l+1):
        # 각 인데스 돌기
        for i in range(l):
            result.append(sum(elements[i:i+limit]))
    return len(list(set(result)))

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int solution(int[] elements) {
        int size = elements.length + elements.length;

        int[] arr = new int[size];
        for (int i = 0; i < elements.length; i++) {
            arr[i] = elements[i];
            arr[i + elements.length] = elements[i];
        }

        Set<Integer> set = new HashSet<>();
        for (int i = 1; i <= elements.length; i++) {
            for (int j=0; j < elements.length; j++) {
                int sum = 0;
                for (int k = j; k < j + i; k++) {
                    sum += arr[k];
                }
                set.add(sum);
            }
        }
        
        return set.size();
    }
}